### PR TITLE
#162171009 Eslint configuration

### PR DESCRIPTION
**Pull request by edogbosunny**
#### What does this PR do?
It configures eslint to the project
#### Description of Task to be completed?
- Install eslint library modules in the project
- write eslint configurations to a new .eslintrc file
- Write a script to fix syntax errors
#### How should this be manually tested?
npm run lint-fix
#### Any background context you want to provide?
Nil
#### What are the relevant pivotal tracker stories?
[Finishes #162171009]
#### Screenshots (if appropriate)
Nil
#### Questions:
Nil

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+    "root": true,
+  "extends": "airbnb-base",
+  "env": {
+    "node": true,
+    "es6": true,
+    "mocha": true
+  },
+  "rules": {
+    "one-var": 0,
```
Comment (Chidi3b3r3) [inline .eslintrc]: one-var should not be turned off

Comment (Chidi3b3r3) [reply .eslintrc]: So you actually do not need to define it

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "one-var-declaration-per-line": 0,
```
Comment (Chidi3b3r3) [inline .eslintrc]: Take away this also

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    "new-cap": 0,
```
Comment (Chidi3b3r3) [inline .eslintrc]: Remove this. We need this to be true

File: .eslintrc [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    "consistent-return": 0,
```
Comment (Chidi3b3r3) [inline .eslintrc]: We need a consistent return statement. Please turn this on

File: .eslintrc [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    "no-param-reassign": 0,
```
Comment (Chidi3b3r3) [inline .eslintrc]: Set this to throw a warning

File: index.js [hunk 6]
```diff
@@ -1,93 +1,95 @@
...
-    express = require("express"),
-    bodyParser = require("body-parser"),
-    session = require("express-session"),
-    cors = require("cors"),
-    passport = require("passport"),
-    errorhandler = require("errorhandler"),
-    mongoose = require("mongoose");
-
-const isProduction = process.env.NODE_ENV === "production";
+const fs = require('fs'),
```
Comment (Chidi3b3r3) [inline index.js]: Each import should be defined in separate lines please

**Comment by Chidi3b3r3** [comment]
Ensure each affected file is converted to ES6 and do read up on the rules you've added to know exactly what is being turned off.

File: package.json [hunk 7]
```diff
@@ -4,6 +4,7 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
+    "lint-fix": "eslint --fix ./ --ext .js --color",
```
Comment (Chidi3b3r3) [inline package.json]: Please change the script to lint:fix

Comment (edogbosunny) [reply package.json]: Thank you for your feedback. This has been corrected

File: .eslintrc (hunk 1 above)
Comment (edogbosunny) [reply .eslintrc]: Thank you for your feedback. This has been corrected

File: .eslintrc [hunk 8]
```diff
@@ -0,0 +1,33 @@
+{
+    "root": true,
```
[the remark this answers is not in the document]
Comment (edogbosunny) [reply .eslintrc]: Thank you for your feedback. I am on it

File: .eslintrc (hunk 1 above)
Comment (edogbosunny) [reply .eslintrc]: Thank you for your feedback. I am on it

File: .eslintrc (hunk 5 above)
Comment (edogbosunny) [reply .eslintrc]: Thank you for your feedback. I am on it

**Pull request merged**